openslides_backend/action/actions/projector/next.py

```diff
@@ -0,0 +1,77 @@
+from typing import Any, Dict
+
+from ....models.models import Projector
+from ....services.datastore.commands import GetManyRequest
+from ....shared.patterns import Collection, FullQualifiedId
+from ...generics.update import UpdateAction
+from ...util.default_schema import DefaultSchema
+from ...util.register import register_action
+from ...util.typing import ActionPayload
+
+
+@register_action("projector.next")
+class ProjectorNext(UpdateAction):
+    """
+    Action to get to the next projection.
+    """
+
+    model = Projector()
+    schema = DefaultSchema(Projector()).get_update_schema()
+
+    def get_updated_instances(self, payload: ActionPayload) -> ActionPayload:
+        for instance in payload:
+            projector = self.datastore.get(
+                FullQualifiedId(self.model.collection, instance["id"]),
+                [
+                    "current_projection_ids",
+                    "preview_projection_ids",
+                    "history_projection_ids",
+                ],
+            )
+            if not projector.get("preview_projection_ids"):
+                continue
+            current_projections = []
+            if projector.get("current_projection_ids"):
+                gmr = GetManyRequest(
+                    Collection("projection"),
+                    projector["current_projection_ids"],
+                    ["id", "stable"],
+                )
+                result = self.datastore.get_many([gmr])
+                current_projections = list(
+                    result.get(Collection("projection"), {}).values()
+                )
+            new_current_projection_ids = [
+                projection["id"]
+                for projection in current_projections
+                if self.is_stable(projection)
+            ]
+            new_history_projection_ids = projector.get("history_projection_ids", []) + [
```

Comment:
I think the `weight` handling is missing.

Reply:
(See https://github.com/OpenSlides/OpenSlides/wiki/Projection#preview-and-history)

Reply:
Add the weight handling and update the test.